src/pages/api/tutorials-map.ts

```diff
@@ -0,0 +1,35 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { getTutorialSlug } from 'views/collection-view/helpers'
+import { getAllTutorials } from 'lib/learn-client/api/tutorial'
+import moize, { Options } from 'moize'
+
+export default async function tutorialsMapHandler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const mapData = await cachedGenerateTutorialMap()
```

Comment:
The intent here is that we only want the `getAllTutorials` call to hit once (it's quite large), and then cache that map for subsequent requests. 

I first tried to memoize the function within the plugin definition file itself. But with ISR, it didn't seem to persist there properly. Memoizing via this API route is, to my testing knowledge, behaving as expected. For a single deploy, the `getAllTutorials` function is called on the first tutorial page hit, and then the cached map is served on subsequent hits. Check the vercel function logs while digging in the preview to validate.

Reply:
<img width="1126" alt="Screen Shot 2022-04-18 at 2 30 24 PM" src="https://user-images.githubusercontent.com/36613477/163881040-ee1fda9d-065b-4ddf-8b08-25d11474b0da.png">

Reply:
These run on AWS lambdas, so there's not necessarily a guarantee that we'll get the same lambda instance on subsequent invocations. If that does happen, the in-memory caching wouldn't do anything for us.

We could set some caching headers to have this API route actually be cached by Vercel ([ref](https://vercel.com/docs/concepts/functions/edge-caching))

Code example: https://github.com/hashicorp/mktg-content-workflows/blob/main/api/assets.ts#L80

Reply:
Oh great! Okay I'll check this out